assets/js/planning-update.js

```diff
@@ -122,23 +148,82 @@ function checkLastUpdate(forceUpdate) {
+  window.getSelection().removeAllRanges();
+  const isChecked = $currentClickedTd.hasClass('checked');
+
+  const minTbodyIndex = Math.min($lastClickedTd.closest('tbody').index(), $currentClickedTd.closest('tbody').index());
+  const maxTbodyIndex = Math.max($lastClickedTd.closest('tbody').index(), $currentClickedTd.closest('tbody').index());
+  const tdFrom = Math.min(Date.parse($lastClickedTd.data('from')), Date.parse($currentClickedTd.data('from')));
+  const tdTo = Math.max(Date.parse($lastClickedTd.data('to')), Date.parse($currentClickedTd.data('to')));
+
+  // default case : maxTbodyIndex === minTbodyIndex
+  let minTrIndex = Math.min($lastClickedTd.closest('tr').index(), $currentClickedTd.closest('tr').index());
+  let maxTrIndex = Math.max($lastClickedTd.closest('tr').index(), $currentClickedTd.closest('tr').index());
+  if (maxTbodyIndex !== minTbodyIndex) {
+    if ($lastClickedTd.closest('tbody').index() === minTbodyIndex) {
+      minTrIndex = $lastClickedTd.closest('tr').index();
+    } else if ($currentClickedTd.closest('tbody').index() === minTbodyIndex) {
+      minTrIndex = $currentClickedTd.closest('tr').index();
+    }
+
+    if ($lastClickedTd.closest('tbody').index() === maxTbodyIndex) {
+      maxTrIndex = $lastClickedTd.closest('tr').index();
+    } else if ($currentClickedTd.closest('tbody').index() === maxTbodyIndex) {
+      maxTrIndex = $currentClickedTd.closest('tr').index();
+    }
+  }
+
+  const handleTbody = function (i, tbody) {
+    const currentTbodyIndex = $(tbody).index();
+
+    $(tbody)
+      .find('tr')
+      .filter((i, tr) => {
+        const trIndex = $(tr).index();
+        if (minTbodyIndex !== maxTbodyIndex) {
+          if (currentTbodyIndex === minTbodyIndex) {
+            return trIndex >= minTrIndex;
+          } else if (currentTbodyIndex === maxTbodyIndex) {
+            return trIndex <= maxTrIndex;
+          }
+
+          return true;
+        }
+
+        return trIndex >= minTrIndex && trIndex <= maxTrIndex;
+      })
+      .each(handleTr);
+  };
+
+  const handleTr = function (i, tr) {
+    $(tr)
+      .find('td')
+      .filter((i, td) => Date.parse($(td).data('from')) >= tdFrom && Date.parse($(td).data('to')) <= tdTo)
+      .each(function () {
+        if (isChecked !== $(this).hasClass('checked')) {
+          selectTableBox($(this));
+        }
+      });
+  };
+
+  $planning
+    .find('tbody.item-rows')
+    .filter((index, currentTbody) => $(currentTbody).index() >= minTbodyIndex && $(currentTbody).index() <= maxTbodyIndex)
+    .each(handleTbody);
+}
+
 $(document).ready(function () {
   const $planning = $('.planning');
 
+  $planning.on('click', '.slot-box', handleSlotBoxClick);
+  $(document).on('keydown keyup', handleShiftInput);
+
   let urlParams = new URLSearchParams(window.location.search);
   if (urlParams.has('scrollTop')) {
     $(window).scrollTop(urlParams.get('scrollTop'));
   }
 
-  $planning.on('click', '.slot-box input:checkbox', function (e) {
```

Comment:
can you revert this line ? Without it we cannot unselect a box by clicking on the checkbox